instructor/dsl/iterable.py

```diff
@@ -79,7 +79,12 @@ def extract_json(
     ) -> Generator[str, None, None]:
         for chunk in completion:
             try:
-                if chunk.choices:
+                if mode == Mode.ANTHROPIC_JSON:
+                    if json_chunk := chunk.delta.text:
+                        yield json_chunk
+                if mode == Mode.ANTHROPIC_TOOLS:
+                    yield chunk.model_extra.get("delta", "").get("partial_json", "")
```

Comment:
```suggestion
                    yield chunk.model_extra.get("delta", {}).get("partial_json", "")
```

should it be {} since `"".get` would error?